### PR TITLE
MCOL-4116 add long LONGDOUBLE support to update

### PR DESCRIPTION
**Pull request by davidhall586**

### CODE REVIEW

File: writeengine/server/we_dmlcommandproc.cpp [hunk 1]
```diff
@@ -3121,6 +3121,20 @@ uint8_t WE_DMLCommandProc::processUpdate(messageqcpp::ByteStream& bs,
                             break;
                         }
 
+                        case CalpontSystemCatalog::LONGDOUBLE:
+                        {
+                            double dll = row.getDoubleField(fetchColPos);
```
Comment (tntnatbry) [inline writeengine/server/we_dmlcommandproc.cpp]: Shouldn't this be long double in the rest of this case statement?

File: writeengine/server/we_dmlcommandproc.cpp [hunk 2]
```diff
@@ -3121,6 +3121,20 @@ uint8_t WE_DMLCommandProc::processUpdate(messageqcpp::ByteStream& bs,
...
+
+                            if (dll == std::numeric_limits<double>::infinity())
+                                continue;
+
+                            ostringstream os;
+                            //@Bug 3350 fix the precision.
+                            os << setprecision(16) << dll;
```
Comment (tntnatbry) [inline writeengine/server/we_dmlcommandproc.cpp]: I see setprecision(16) for the double case. Is this also accurate for long double?

**Review by tntnatbry — CHANGES_REQUESTED** [review]

File: writeengine/server/we_dmlcommandproc.cpp (hunk 1 above)
Comment (davidhall586) [reply writeengine/server/we_dmlcommandproc.cpp]: blech. Of course.

File: writeengine/server/we_dmlcommandproc.cpp (hunk 2 above)
Comment (davidhall586) [reply writeengine/server/we_dmlcommandproc.cpp]: I'm not sure. I set it to 19. Hopefully that will be OK. 7 for FLOAT above also seems strange.

Comment (tntnatbry) [reply writeengine/server/we_dmlcommandproc.cpp]: https://en.cppreference.com/w/cpp/types/numeric_limits/digits10

Precision for float, double, and long double is 6, 15, and 18 respectively. We are setting them to one higher than the standard. I believe this is happening throughout the code base.

**Review by tntnatbry — APPROVED** [review]

**Pull request merged**